app/src/androidTest/java/org/mozilla/reference/browser/helpers/SessionLoadedIdlingResource.kt

```diff
@@ -31,8 +31,12 @@ class SessionLoadedIdlingResource : IdlingResource {
         return if (session?.loading == true) {
             false
         } else {
-            invokeCallback()
-            true
+            if (session?.progress == 100) {
+                invokeCallback()
+                true
+            } else {
+                false
+            }
```

Comment:
Hey @jayeshsolanki93 

I'm Aaron from Mozilla working on automation on the Fenix project. I stumbled upon this through Github search because we are running into the exact same issue now. I believe the tests need to wait for the Gecko session to be idle before continuing. I was wondering if you got this to work and if it's working recently? I tried adding this addition but I'm not sure entirely if this idling resource is even doing anything?

I noticed the registration is in the activity launch https://github.com/cliqz/daisy/blob/master/app/src/androidTest/java/org/mozilla/reference/browser/helpers/BrowserActivityTestRule.kt - do you know if this works? (I didn't write this originally either). It seems like we have the same issue and I've been struggling to find a solution.

Reply:
Hi Aaron,

 the mechanism is working for us, but we are not aligned with the android-components (we use version 29.0.0 currently).